### PR TITLE
[WFLY-18163] Add org.wildfly.extension.elytron.jaas-realm module exclusion 

### PR DESCRIPTION
**Pull request by skyllarr**
https://issues.redhat.com/browse/WFLY-18163

This PR is required so that the CI on https://github.com/wildfly/wildfly-core/pull/5587 passes

This is a **feature** https://issues.redhat.com/browse/WFLY-18163

### CODE REVIEW

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.io.File;
+import java.net.URL;
+
+import static javax.servlet.http.HttpServletResponse.SC_FORBIDDEN;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
+import static javax.servlet.http.HttpServletResponse.SC_UNAUTHORIZED;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class JaasCustomRealmTest {
```
Comment (OndrejKotek) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java:53]: Fails on Windows:

```
Failed to execute line 'module remove --name=jaasLoginModule'
[org.jboss.as](http://org.jboss.as/).cli.CommandLineException: java.util.concurrent.ExecutionException: [org.jboss.as](http://org.jboss.as/).cli.CommandLineException: Failed to delete testsuite\integration\elytron\target\wildfly\modules\jaasLoginModule\main\testJaas.jar
 at [org.jboss.as](http://org.jboss.as/).cli.impl.CommandContextImpl.execute([CommandContextImpl.java:944](http://commandcontextimpl.java:944/))
 at [org.jboss.as](http://org.jboss.as/).cli.impl.CommandContextImpl.handleLegacyCommand([CommandContextImpl.java:1945](http://commandcontextimpl.java:1945/))
 at [org.jboss.as](http://org.jboss.as/).cli.impl.CommandContextImpl.handle([CommandContextImpl.java:881](http://commandcontextimpl.java:881/))
 at [org.jboss.as.test.integration.management](http://org.jboss.as.test.integration.management/).util.CLIWrapper.sendLine([CLIWrapper.java:218](http://cliwrapper.java:218/))
 at [org.jboss.as.test.integration.management](http://org.jboss.as.test.integration.management/).util.CLIWrapper.sendLine([CLIWrapper.java:244](http://cliwrapper.java:244/))
 at org.wildfly.test.integration.elytron.realm.JaasCustomRealmTest.cleanUp([JaasCustomRealmTest.java:119](http://jaascustomrealmtest.java:119/))
...
```

Comment (OndrejKotek) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java:53]: Fails with Galleon profile:

```
Failed to execute line '/subsystem=elytron/custom-realm=customJaasRealm:add(module=elytronExtensionModule,class-name=org.wildfly.extension.elytron.JaasCustomSecurityRealmWrapper,configuration={entry=Entry1,module=jaasLoginModule,callback-handler=org.wildfly.test.integration.elytron.realm.TestCallbackHandler,path=testsuite/integration/elytron/target/test-classes/org/wildfly/test/integration/elytron/realm/jaas-login.config})'
[org.jboss.as](http://org.jboss.as/).cli.CommandLineException: {
    "outcome" => "failed",
    "failure-description" => {"WFLYCTL0080: Failed services" => {"[org.wildfly.security](http://org.wildfly.security/).security-realm.customJaasRealm" => "java.lang.ClassNotFoundException: org.wildfly.extension.elytron.JaasCustomSecurityRealmWrapper from [Module \"elytronExtensionModule\" from local module loader @8ad6665 (finder: local module finder @30af5b6b (roots: testsuite/integration/elytron/target/wildfly/modules,testsuite/integration/elytron/target/wildfly/modules/system/layers/base,testsuite/integration/elytron/target/modules))]
    Caused by: java.lang.ClassNotFoundException: org.wildfly.extension.elytron.JaasCustomSecurityRealmWrapper from [Module \"elytronExtensionModule\" from local module loader @8ad6665 (finder: local module finder @30af5b6b (roots: testsuite/integration/elytron/target/wildfly/modules,testsuite/integration/elytron/target/wildfly/modules/system/layers/base,testsuite/integration/elytron/target/modules))]"}},
    "rolled-back" => true
}
 at [org.jboss.as](http://org.jboss.as/).cli.handlers.OperationRequestHandler.handle([OperationRequestHandler.java:95](http://operationrequesthandler.java:95/))
 at [org.jboss.as](http://org.jboss.as/).cli.impl.CommandContextImpl.handleOperation([CommandContextImpl.java:1921](http://commandcontextimpl.java:1921/))
 at [org.jboss.as](http://org.jboss.as/).cli.impl.CommandContextImpl.handle([CommandContextImpl.java:876](http://commandcontextimpl.java:876/))
 at [org.jboss.as.test.integration.management](http://org.jboss.as.test.integration.management/).util.CLIWrapper.sendLine([CLIWrapper.java:218](http://cliwrapper.java:218/))
 at [org.jboss.as.test.integration.management](http://org.jboss.as.test.integration.management/).util.CLIWrapper.sendLine([CLIWrapper.java:244](http://cliwrapper.java:244/))
 at org.wildfly.test.integration.elytron.realm.JaasCustomRealmTest.prepareSecurityRealmThatUsesJaasRealmViaCustomRealm([JaasCustomRealmTest.java:85](http://jaascustomrealmtest.java:85/))
...
```

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Test
+    public void testUser1Allowed(@ArquillianResource URL url) throws Exception {
+        final URL servletUrl = new URL(url.toExternalForm() + "role1");
+        Utils.makeCall(servletUrl.toURI(), SC_UNAUTHORIZED); // assert that auth is required
+        Utils.makeCallWithBasicAuthn(servletUrl, "user1", "password1", SC_OK);
+    }
+
+    @Test
+    public void testUser2NotAllowed(@ArquillianResource URL url) throws Exception {
```
Comment (OndrejKotek) [inline testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java:107]: > `testUser2NotAllowed`

-> `testRoleRequired` ?

However, `testUser2NotAllowed` would be also useful.

File: testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java (hunk 1 above)
Comment (Skyllarr) [reply testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/realm/JaasCustomRealmTest.java:53]: Fixed

**Comment by Skyllarr** [comment]
@OndrejKotek Thank you for review!

**Review by fjuma — APPROVED** [review]
Thanks @Skyllarr! Approving these changes. We'll need to discuss how to coordinate getting https://github.com/wildfly/wildfly-core/pull/5587 merged once pre-checked since CI passing will depend on the exclusions that are being added in this PR.

**Review by fjuma — APPROVED** [review]

**Comment by Skyllarr** [comment]
I have split this PR so that the CI passes on this one. The test was moved to the new PR https://github.com/wildfly/wildfly/pull/17417 .

It might be easier to coordinate the merges with https://github.com/wildfly/wildfly-core/pull/5587

**Comment by Skyllarr** [comment]
Hi @yersan , I was trying to make HostExcludesTestCase work, so I created WFLY-18764 issue and included its commit in this PR. But now I can't get pass the below error in "Thin Server Build":

Invalid value WildFly30.0 for host-release; legal values are [WildFly23.0, WildFly24.0, WildFly25.0, WildFly26.0, WildFly20.0, WildFly21.0, WildFly22.0, WildFly27.0, WildFly28.0, WildFly29.0, WildFly12.0, WildFly13.0, WildFly14.0, WildFly15.0, WildFly10.0, WildFly10.1, WildFly11.0, EAP6.3, EAP7.2, EAP6.2, EAP7.1, EAP7.0, WildFly16.0, WildFly17.0, EAP7.4, WildFly18.0, EAP6.4, EAP7.3, WildFly19.0]"}}}
	at org.wildfly.galleon.plugin.config.generator.BaseConfigGenerator.doHandle(BaseConfigGenerator.java:144)
	at org.wildfly.galleon.plugin.config.generator.BaseConfigGenerator.endBatch(BaseConfigGenerator.java:162)
	at org.wildfly.galleon.plugin.config.generator.ForkedConfigGenerator.executeScript(ForkedConfigGenerator.java:63)
	at org.wildfly.galleon.plugin.config.generator.ForkedConfigGenerator.forkedForEmbedded(ForkedConfigGenerator.java:45)
	at org.wildfly.galleon.plugin.server.ForkedProcessRunner.main(ForkedProcessRunner.java:56)

do you happen to know why the WildFly30.0 is not an allowed value? Thanks a lot!

**Comment by yersan** [comment]
Hi @Skyllarr , the changes in the HostExclusion test case and domain exclusions look good, the issue you are facing is that we have not updated WildFly Core to the management API used by WildFly 31. This will be done at https://issues.redhat.com/browse/WFCORE-6620, we plan to release it this Wednesday at most.

**Comment by Skyllarr** [comment]
@yersan I see, thank you very much for explanation and for looking at my HostExcludesTestCase changes!

**Review by darranl — APPROVED** [review]

File: testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java [hunk 3]
```diff
@@ -53,6 +53,7 @@ public abstract class LayersTestBase {
             "org.hornetq.client",
             // TODO we need to add an xts layer
             "org.jboss.as.xts",
+            "org.wildfly.extension.elytron.jaas-realm",
```
Comment (darranl) [inline testsuite/shared/src/main/java/org/jboss/as/test/shared/LayersTestBase.java:56]: This PR has needed more additions but in general could it be useful if the LatestTest got some of these items directly from a class in WildFly Core so when changes are made to WildFly Core we reduce the number of round trips to get something merged.

**Comment by yersan** [comment]
Superseded by https://github.com/wildfly/wildfly/pull/17622, already merged

**Pull request closed without merging**